smdebug/tensorflow/keras.py

```diff
@@ -736,6 +741,46 @@ def _prepare_collections_for_tf2(self):
             self._unwrap_model_with_input_output_saver()
         self.prepared_tf2_collections = True
 
+    def _handle_start_step_python_profiling(self, mode, current_step):
+        """
+        This function is called to handle python profiling at the start of a step.
+        :param mode: ModeKeys.TRAIN ModeKeys.EVAL ModeKeys.PREDICT
+        """
+        if self.python_profiler:
+            self.python_profiler.stop_profiling(
+                StepPhase.STEP_START,
+                end_mode=mode_keys_to_python_profile_mode(mode),
+                end_step=current_step,
+            )
+            if self.profiler_config_parser.should_save_metrics(
+                MetricsCategory.PYTHON_PROFILING, current_step
+            ):
+                self.python_profiler.start_profiling(
```

Comment:
Re-evaluate the profiler can accept the profiling config and handle all the cases.

Reply:
Done